app/uk/gov/hmrc/economiccrimelevyregistration/controllers/deregister/DeregisterCheckYourAnswersController.scala

```
@@ -0,0 +1,112 @@
+import uk.gov.hmrc.economiccrimelevyregistration.controllers.actions.deregister.DeregistrationDataRetrievalAction
+import uk.gov.hmrc.economiccrimelevyregistration.controllers.{BaseController, ErrorHandler}
+import uk.gov.hmrc.economiccrimelevyregistration.models.ContactDetails
+import uk.gov.hmrc.economiccrimelevyregistration.models.deregister.Deregistration
+import uk.gov.hmrc.economiccrimelevyregistration.services.EclRegistrationService
+import uk.gov.hmrc.economiccrimelevyregistration.services.deregister.DeregistrationService
+import uk.gov.hmrc.economiccrimelevyregistration.viewmodels.deregister._
+import uk.gov.hmrc.economiccrimelevyregistration.viewmodels.govuk.SummaryListFluency
+import uk.gov.hmrc.economiccrimelevyregistration.views.html.ErrorTemplate
+import uk.gov.hmrc.economiccrimelevyregistration.views.html.deregister.DeregisterCheckYourAnswersView
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist.SummaryList
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class DeregisterCheckYourAnswersController @Inject() (
+  val controllerComponents: MessagesControllerComponents,
+  authorise: AuthorisedActionWithEnrolmentCheck,
+  getDeregistrationData: DeregistrationDataRetrievalAction,
+  eclRegistrationService: EclRegistrationService,
+  deregistrationService: DeregistrationService,
+  appConfig: AppConfig,
+  view: DeregisterCheckYourAnswersView
+)(implicit ec: ExecutionContext, errorTemplate: ErrorTemplate)
+    extends FrontendBaseController
+    with I18nSupport
+    with BaseController
+    with ErrorHandler
+    with SummaryListFluency {
+
+  def onPageLoad(): Action[AnyContent] = (authorise andThen getDeregistrationData).async { implicit request =>
+    val eclReference = request.eclRegistrationReference.getOrElse("")
```

is it okay not to have an eclReference here or should we return an error in this case?

---

If you have no ECL reference then the `getSubscription` will fail so an error will be returned.

---

We shouldn't be using getOrElse here.

Match on request.eclRegistrationReference and handle the error case.